Description page of each category after a person submits a photo

B-Tanonaka opened the pull request:
To view components:
localhost:3000/recycle
localhost:3000/trash
localhost:3000/compost

1. Created src/Components/CategoryDescription.js file that uses the submitted photo's category to display a page with basic information about what to do with their litter.
2. Currently uses dummy data and is not connected to the photo submission system.
3.  Buttons navigate to appropriate page; LearnMore is a sample file that will be adjusted later.
4. Changed the body to display flex to center all components for mobile.

JMStudiosJoe commented on src/App.css:
```diff
@@ -1,3 +1,8 @@
+body {
+  display: flex;
```
for me, at the parent level components such as App, landingPage and ImageCapture page or any others it would be best to use CSS Grid with Grid layout to declare how the page lays out in mobile and in web. CSS Grid is very easy and powerful at the parent level but it can be a little overkill at times when you just want things to align like here. For now though, this should work, just food for thought.

B-Tanonaka replied:
I had not thought about using CSS Grid but I see what you mean and I can switch it. Do you suggest having different grid selections for web and mobile using @media queries?

JMStudiosJoe replied:
yes, likely using media queries, can either add here or in a separate PR

B-Tanonaka replied:
I've made the changes in this PR so that the leaderboard and category description are centered with CSS Grid. Right now theres duplicated code for the media queries since each component has its own css file but it can reduced if all of the css files are later combined.

JMStudiosJoe commented on src/App.js:
```diff
@@ -14,6 +16,10 @@ const App = () => {
         <Route path='/' element={<LandingPage />} />
         <Route path='/capture' element={<CameraCapture />} />
         <Route path='/leaderboard' element={<Leaderboard />} />
+        <Route path='/recycle' element={<CategoryDescription type={categoryData.recycle}/>} />
```
all different routes? not on one page for clarity?

B-Tanonaka replied:
Good point. I've changed it to a single page and will feed in the type based on the item using the photo's metadata.

JMStudiosJoe commented on src/CategoryDescription.css:
```diff
@@ -0,0 +1,63 @@
+:root {
```
I am not too sure here but does this need to be prefixed by the .category-container class or added to a main root css file for colors?

B-Tanonaka replied:
I pulled the colors from the Figma file so I think everyone can use it. Moving this to App.css.

JMStudiosJoe commented on src/CategoryDescription.css:
```diff
@@ -0,0 +1,63 @@
+  font-size: 1.2rem;
+}
+
+.category-container .category-buttons button:hover {
```
great css, might be little extra code but it is very clear and scoped to this file great work.

B-Tanonaka replied:
Thank you. And thanks for reviewing this PR so quickly!

JMStudiosJoe commented on src/CategoryDescription.css:
```diff
@@ -0,0 +1,69 @@
+@media only screen and (max-width: 767px) {
+  .category-container {
```
grid-template-area might also be something to look into, it is very useful for declaring where everything goes instead of these ratios that can still raise questions. just something to note

JMStudiosJoe commented on src/Components/LearnMore.js:
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
```
lets add test files for this component.

JMStudiosJoe commented on src/Components/CategoryDescription.js:
```diff
@@ -0,0 +1,26 @@
+import EggIcon from '@mui/icons-material/Egg';
+import '../CategoryDescription.css'
+
+export default function CategoryDescription ({ type }) {
```
lets add test files for this component.

JMStudiosJoe commented on src/App.js:
```diff
@@ -1,21 +1,27 @@
+import LearnMore from './Components/LearnMore';
 
 const App = () => {
   return (
```
this component is very high up in the root and difficult to unti test out at times so no test needed for this component.

JMStudiosJoe reviewed (changes requested):
just some tests to add please and then will be good to go.

JMStudiosJoe commented:
examples of how to write some snapshot tests as UI unit test are here https://github.com/codeforsanjose/litter.ai/blob/main/src/Components/__tests__/litter.test.js
and will need to be moved as well soon closer to the root or along with every component we make to keep things concise.

B-Tanonaka commented:
@JMStudiosJoe I wrote tests for these two components but I put them on a new branch that has the leaderboard connected to the backend so I'm going to close this PR and open a new one.

B-Tanonaka commented:
@JMStudiosJoe I previously closed this PR because I made the changes on a new branch and thought it would be easier to open up a new PR but I was able to merge my branches.

This version uses data from the API that was created and it needs to have it running on port 3001 for it to currently work. Will refactor once the API is hosted or on the same repo.

Changes include:
1. Added tests for the Leaderboard and SuccessfulSubmission (previously CategoryDescription) pages.
2. Added react-select menu in the leaderboard to change the data based on category.

JMStudiosJoe reviewed (approved):
some changes will need to be made in the leaderboard urls but lets focus on a cleanup pr after this one
